pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/RangeIndexHandler.java

```diff
@@ -48,13 +52,18 @@
   private final SegmentMetadata _segmentMetadata;
   private final SegmentDirectory.Writer _segmentWriter;
   private final Set<String> _columnsToAddIdx;
+  private final int _rangeIndexVersion;
 
   public RangeIndexHandler(File indexDir, SegmentMetadata segmentMetadata, IndexLoadingConfig indexLoadingConfig,
       SegmentDirectory.Writer segmentWriter) {
     _indexDir = indexDir;
     _segmentMetadata = segmentMetadata;
     _segmentWriter = segmentWriter;
     _columnsToAddIdx = new HashSet<>(indexLoadingConfig.getRangeIndexColumns());
+    // guard against null table config
+    _rangeIndexVersion = Optional.ofNullable(indexLoadingConfig.getTableConfig())
```

Comment:
We can add the `_rangeIndexVersion` as a field in the `IndexLoadingConfig`, and allow overriding it with table config. That's the way we handle other index related configs